components/patina_smbios/src/service.rs

```diff
@@ -101,22 +101,82 @@ impl<'a> Iterator for SmbiosRecordsIter<'a> {
     }
 }
 
-/// SMBIOS service providing all SMBIOS operations
+/// Object-safe trait for SMBIOS service operations
```

Comment:
Could you rename this trait to remove `Service`? I thought we had documentation (I cannot find it, I will update) we try and avoid the word "Config" in the struct name for configs, and "Service" in the trait name for services because ultimately the user requests them via `Config<MyConfig>` and `Service<dyn MyService>` which gets a bit repetitive.

My suggestion is to rename the trait to `Smbios` and the implementor to something else. But it can be somethign else. Your choice.